app/src/main/java/com/randomappsinc/studentpicker/backupandrestore/backup/BackupDataFragment.kt

```diff
@@ -0,0 +1,149 @@
+package com.randomappsinc.studentpicker.backupandrestore.backup
+
+import android.Manifest
+import android.app.Activity
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import com.randomappsinc.studentpicker.R
+import com.randomappsinc.studentpicker.databinding.BackupDataBinding
+import com.randomappsinc.studentpicker.utils.PermissionUtils
+import com.randomappsinc.studentpicker.utils.PreferencesManager
+import com.randomappsinc.studentpicker.utils.TimeUtils
+import com.randomappsinc.studentpicker.utils.UIUtils
+
+private const val WRITE_EXTERNAL_STORAGE_CODE = 1
+private const val WRITE_BACKUP_FILE_CODE = 350
+
+class BackupDataFragment : Fragment() {
+
+    private var _binding: BackupDataBinding? = null
+    private val binding get() = _binding!!
+
+    private lateinit var preferencesManager: PreferencesManager
+
+    override fun onCreateView(
+            inflater: LayoutInflater,
+            container: ViewGroup?,
+            savedInstanceState: Bundle?
+    ): View? {
+        _binding = BackupDataBinding.inflate(inflater, container, false)
+        preferencesManager = PreferencesManager(context)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        backupDataClicked()
```

Comment:
```suggestion
        setUpBackupButtonOnClickListener()
```
The current name seems like something that would fire after the button has been clicked.